src/pages/workspaces/workspace/analysis/Analyses.ts

```diff
@@ -230,163 +251,143 @@ const AnalysisCard = ({
+  span({ style: { fontWeight: 'bold', marginLeft: '0.5ch' } }, ['This may take a few minutes.'])
+])
+
+export type DisplayAnalysisFile = AnalysisFile & {
+  application: ToolLabel
+}
+
+export const decorateAnalysisFiles = (rawAnalyses: AnalysisFile[]): DisplayAnalysisFile[] => {
+  const notebooks: AnalysisFile[] = _.filter(({ name }) => _.includes(getExtension(name), runtimeTools.Jupyter.ext), rawAnalyses)
+  const rAnalyses: AnalysisFile[] = _.filter(({ name }) => _.includes(getExtension(name), runtimeTools.RStudio.ext), rawAnalyses)
+
+  //we map the `toolLabel` corresponding header label, which simplifies the table sorting code
+  const enhancedNotebooks: DisplayAnalysisFile[] = _.map(file => ({ application: tools.Jupyter.label, ...file }), notebooks)
+  const enhancedRmd: DisplayAnalysisFile[] = _.map(file => ({ application: tools.RStudio.label, ...file }), rAnalyses)
+
+  const tempAnalyses: DisplayAnalysisFile[] = _.concat(enhancedNotebooks, enhancedRmd)
+  return _.reverse(_.sortBy(tableFields.lastModified, tempAnalyses))
+}
+
+export const getUniqueFileName = (originalName: string, existingFileNames: FileName[]): FileName => {
+  const name: FileName = originalName as FileName
+  let resolvedName: FileName = name
+  let c = 0
+  while (_.includes(resolvedName, existingFileNames)) {
+    resolvedName = `${getDisplayName(name)}_${++c}.${getExtension(name)}` as FileName
+  }
+  return resolvedName
+}
+
+export interface AnalysesData {
+  apps: App[]
+  refreshApps: () => Promise<void>
+  runtimes: Runtime[]
+  refreshRuntimes: () => Promise<void>
+  appDataDisks: PersistentDisk[]
+  persistentDisks: PersistentDisk[]
+}
+
+export interface AnalysesProps {
+  workspace: WorkspaceWrapper
+  analysesData: AnalysesData
+  onRequesterPaysError: () => void
+  storageDetails: StorageDetails
+}
+
+export interface SortOrderInfo {
+  field: string
+  direction: string
+}
+
+export const BaseAnalyses: React.FC<AnalysesProps> = ({
+  workspace,
   analysesData: { apps, refreshApps, runtimes, refreshRuntimes, appDataDisks, persistentDisks },
   storageDetails: { googleBucketLocation, azureContainerRegion },
   onRequesterPaysError
-}, _ref) => {
-  const [renamingAnalysisName, setRenamingAnalysisName] = useState(undefined)
-  const [copyingAnalysisName, setCopyingAnalysisName] = useState(undefined)
-  const [deletingAnalysisName, setDeletingAnalysisName] = useState(undefined)
-  const [exportingAnalysisName, setExportingAnalysisName] = useState(undefined)
-  const [sortOrder, setSortOrder] = useState(() => getLocalPref(KEY_ANALYSES_SORT_ORDER) || defaultSort.value)
-  const enableJupyterLabPersistenceId = `${namespace}/${workspaceName}/enableJupyterLabGCP`
-  const [enableJupyterLabGCP, setEnableJupyterLabGCP] = useState(() => getLocalPref(enableJupyterLabPersistenceId) || false)
+}: AnalysesProps, _ref) => {
+  const [renamingAnalysisName, setRenamingAnalysisName] = useState<AbsolutePath | undefined>(undefined)
```

Comment:
`@types/react` provides a convenience helper for this case.

```ts
const [state, setState] = useState<State>()
```
is the same as
```ts
const [state, setState] = useState<State | undefined>(undefined)
```

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/584a2f24228f524d696886617126a85ae26ed6ca/types/react/index.d.ts#L899-L906